### ISSUE TITLE
Can't merge non-disjoint spans. '🇸' is already part of tokens to merge.

### ISSUE THREAD

**Issue by jjhbw** [ines/spacymoji#5]
First off, thanks for this amazingly useful library! Really takes the sting out of working with social media texts.

I found something that looks like a bug occurring with multi-codepoint emoji's.
Repro case:
```python
import en_core_web_sm
from spacymoji import Emoji

nlp = en_core_web_sm.load()
em = Emoji(nlp)

em(nlp(u"🇺🇸🇦🇷"))
```

This raises the exception shown in the title.
Initializing `Emoji` with `merge_spans=False` circumvents this.

Let me know if I can help!

**Comment by scribu** [comment]
Started seeing the same issue. Used to work fine a few months ago. 🤔 

Using spacy 2.1.6 + spacymoji 2.0

**Comment by scribu** [comment]
Using spacy 2.1.6 + spacymoji 2.0

**Comment by scribu** [comment]
#8 is my attempt at fixing this. Please try it out and let me know if you encounter other edge cases.

**Comment by SiirRandall** [comment]
Hello. I am also getting the Can't merge non-disjoint spans. '🇦' is already part of tokens to merge. Using the latest off of pip.

**Comment by juiceboost** [comment]
please fix 😢

**Comment by cbjrobertson** [comment]
I have installed #8 from source and am still experiencing this issue. My `spacy` version is `2.1.8` and my load/use case is: 

```
text = '"Tonight Peru and Colombia will go to battle. Let's see who wins the war. 🇵🇪🇨🇴"'
import spacy
import spacymoji
nlp = spacy.load('en_core_web_sm')
emoji = spacymoji.Emoji(nlp)
nlp.add_pipe(emoji, first=True)
doc = nlp(text)
>>>ValueError: [E102] Can't merge non-disjoint spans. '🇪' is already part of tokens to merge.
```

**Comment by roma-glushko** [comment]
Got the same issue 😪 
Looks like not only a sequence of emojis can trigger it but also a single emoji like '👨‍👩‍👧‍👦':

```python
import spacy
from spacymoji import Emoji

nlp = spacy.load("en_core_web_sm")
em = Emoji(nlp)

## case 1
em(nlp(u"👨‍👩‍👧‍👦"))
## case 2
em(nlp(u"🇩🇪 🇬🇧 🇳🇱 🇺🇦 🇺🇲 🇸🇪"))

```

**Comment by bogdankostic** [comment]
Are there plans to fix this issue anytime soon?

**Comment by andaziemele** [comment]
Hi, having the same issue here with 👨.

**Comment by abushoeb** [comment]
Does anyone have an update on this?

**Comment by coreyryanhanson** [comment]
It appears the code hasn't been updated in two years. It worked fine for me by creating a new Emoji class and updating with modifications to the code.

See the pull request in the thread by @scribu . It fixed the problem for me. I wonder why it hasn't been brought into the main project yet.

**Comment by i-like-the-idea** [comment]
Can someone merge the fix please?

**Comment by polm** [comment]
Hey folks, sorry for not getting to this earlier but it should be resolved by #13 and fixed in recent releases.
